extensions/wikipedia/src/search-page.tsx

```diff
@@ -1,48 +1,54 @@
-import { ActionPanel, CopyToClipboardAction, Icon, List, OpenInBrowserAction, PushAction } from "@raycast/api";
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCachedState, usePromise } from "@raycast/utils";
 import { useState } from "react";
-import { encodeTitle, useWikipediaPageData, useWikipediaSearch } from "./wikipedia";
 import ShowDetailsPage from "./show-details-page";
+import { findPagesByTitle, getPageData } from "./wikipedia";
 
 export default function SearchPage() {
   const [search, setSearch] = useState("");
-  const { data: titles, isValidating } = useWikipediaSearch(search);
+  const [language, setLanguage] = useCachedState("language", "en");
+  const { data, isLoading } = usePromise(findPagesByTitle, [search, language]);
 
   return (
     <List
       throttle
-      isLoading={isValidating}
+      isLoading={isLoading}
       onSearchTextChange={setSearch}
       searchBarPlaceholder="Search pages by name..."
+      searchBarAccessory={
+        <List.Dropdown tooltip="Language" value={language} onChange={setLanguage}>
```

Comment:
Maybe you can use [storeValue property](https://developers.raycast.com/api-reference/user-interface/list#props-1) instead?

Reply:
I've applied that change here: https://github.com/raycast/extensions/pull/3976/commits/59f29803c6129dddfce5a2385efc37817ecfa7bf

Is this the approach you would recommend?